pkg/fanal/analyzer/analyzer.go

```diff
@@ -313,14 +339,24 @@ func (ag AnalyzerGroup) AnalyzeFile(ctx context.Context, wg *sync.WaitGroup, lim
 		return nil
 	}
 
+	// filepath extracted from tar file doesn't have the prefix "/"
+	cleanPath := strings.TrimLeft(filePath, "/")
+
 	for _, a := range ag.analyzers {
 		// Skip disabled analyzers
 		if slices.Contains(disabled, a.Type()) {
 			continue
 		}
 
-		// filepath extracted from tar file doesn't have the prefix "/"
-		if !a.Required(strings.TrimLeft(filePath, "/"), info) {
+		filePatternMatch := false
+		for _, pattern := range ag.filePatterns[a.Type()] {
+			if pattern.MatchString(cleanPath) {
+				filePatternMatch = true
+				break
+			}
+		}
```

Comment:
Having it in a dedicated method seems better.

func (ag AnalyzerGroup) filePatternMatch(analyzerType Type, filePath string) {
	for _, pattern := range ag.filePatterns[analyzerType] {
		if pattern.MatchString(cleanPath) {
			return true
		}
	}
	return false
}